File: 프로그래머스/2/68645. 삼각 달팽이/삼각 달팽이.java
class Solution {
    public int[] solution(int n) {
        int[][] arr = new int[n][n];

        int num = 1;
        int row = -1;
        int col = 0;

        arr[row+1][col] = num;

        for (int i = n; i > 0; i -= 3) {
            for (int j = 0; j < i; j++) {
                arr[++row][col] = num++;
            }
            for (int j = 0; j < i - 1; j++) {
                arr[row][++col] = num++;
            }
            for (int j = 0; j < i - 2; j++) {
                arr[--row][--col] = num++;
            }
        }

        int[] answer = new int[num - 1];
        int idx = 0;
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                if (arr[i][j] != 0) {
                    answer[idx++] = arr[i][j];
                }
            }
        }

        return answer;
    }
}

File: 프로그래머스/2/68645. 삼각 달팽이/삼각 달팽이.py
def solution(n):
    arr = [[0] * n for _ in range(n)]

    num = 1
    row = -1
    col = 0

    arr[row + 1][col] = num

    for i in range(n, 0, -3):
        for j in range(i):
            row += 1
            arr[row][col] = num
            num += 1
        for j in range(i - 1):
            col += 1
            arr[row][col] = num
            num += 1
        for j in range(i - 2):
            row -= 1
            col -= 1
            arr[row][col] = num
            num += 1

    answer = [arr[i][j] for i in range(n) for j in range(n) if arr[i][j] != 0]

    return answer